### PR TITLE
Add 'Index.search' API wrapper.

### PR DESCRIPTION
**Pull request by tseaver**

### CODE REVIEW

File: gcloud/search/index.py [hunk 1]
```diff
@@ -197,3 +197,93 @@ def list_documents(self, max_results=None, page_token=None,
...
+
+        :type query: string
+        :param query: query string (see https://cloud.google.com/search/query).
+
+        :type max_results: int
+        :param max_results: maximum number of zones to return, If not
+                            passed, defaults to a value set by the API.
+
+        :type page_token: string
+        :param page_token: opaque marker for the next "page" of zones. If
```
Comment (dhermes) [inline gcloud/search/index.py]: zones

File: gcloud/search/index.py [hunk 2]
```diff
@@ -197,3 +197,93 @@ def list_documents(self, max_results=None, page_token=None,
...
+                           not passed, the API will return the first page of
+                           zones.
```
Comment (dhermes) [inline gcloud/search/index.py]: zones

File: gcloud/search/index.py [hunk 3]
```diff
@@ -197,3 +197,93 @@ def list_documents(self, max_results=None, page_token=None,
...
+        """Search documents created within this index.
+
+        See:
+        https://cloud.google.com/search/reference/rest/v1/projects/indexes/search
+
+        :type query: string
+        :param query: query string (see https://cloud.google.com/search/query).
+
+        :type max_results: int
+        :param max_results: maximum number of zones to return, If not
```
Comment (dhermes) [inline gcloud/search/index.py]: zones

**Comment by dhermes** [comment]
LGTM pending docstring issues

File: gcloud/search/index.py (hunk 3 above)
Comment (tseaver) [inline gcloud/search/index.py]: 54f7922

File: gcloud/search/index.py (hunk 2 above)
Comment (tseaver) [inline gcloud/search/index.py]: 54f7922

File: gcloud/search/index.py (hunk 1 above)
Comment (tseaver) [inline gcloud/search/index.py]: 54f7922 fixes this one, plus a number of others.

**Pull request merged**